### PR TITLE
EWPP-1299: The oEmbed filter.

### PR DESCRIPTION
**Pull request by upchuk**
Creates the filter for transforming the embed code into the rendered entities.

### CODE REVIEW

**Review by 22Alexandra — APPROVED** [review]

File: tests/src/Functional/EmbedFilterTest.php [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    }
+    $embedded_node = $this->drupalGetNodeByTitle('Embedded node');
+    $content .= '<p data-oembed="https://oembed.ec.europa.eu?url=https%3A//data.ec.europa.eu/ewp/node/' . $embedded_node->uuid() . '"><a href="https://data.ec.europa.eu/ewp/node/' . $embedded_node->uuid() . '">' . $embedded_node->label() . '</a></p>';
+
+    $values = [];
+    $values['type'] = 'page';
+    $values['title'] = 'Test node with all the embeds';
+    $values['body'] = [['value' => $content, 'format' => 'format_with_embed']];
+    $node = $this->drupalCreateNode($values);
+    $this->drupalGet('node/' . $node->id());
```
Comment (22Alexandra) [inline tests/src/Functional/EmbedFilterTest.php:135]: Missing access check test coverage.

**Review by 22Alexandra — APPROVED** [review]

File: src/Plugin/Filter/FilterOembed.php [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    $result->addCacheableDependency($cache);
+
+    $this->replaceNodeContent($node, $output);
+  }
+
+  /**
+   * Converts the body of a \DOMDocument back to an HTML snippet.
+   *
+   * Custom serialize method used to serializes the body part of a \DOMDocument
```
Comment (sergepavle) [inline src/Plugin/Filter/FilterOembed.php:201]:
```suggestion
   * Custom serialize method used to serialize the body part of a \DOMDocument
```

File: tests/src/Functional/EmbedFilterTest.php [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+namespace Drupal\Tests\oe_oembed\Functional;
+
+use Drupal\editor\Entity\Editor;
+use Drupal\file\Entity\File;
+use Drupal\filter\Entity\FilterFormat;
+use Drupal\media\Entity\Media;
+
+/**
+ * Tests the oEmbed media filter.
```
Comment (sergepavle) [inline tests/src/Functional/EmbedFilterTest.php:13]:
```suggestion
 * Tests the oEmbed entities filter.
```

File: tests/src/Functional/EmbedFilterTest.php [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      ],
+    ]);
+    $editor->save();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->createTestEntities();
+  }
+
+  /**
+   * Tests that the media oEmbed filter correctly renders the oEmbed tags.
```
Comment (sergepavle) [inline tests/src/Functional/EmbedFilterTest.php:65]:
```suggestion
   * Tests that the oEmbed filter correctly renders the oEmbed tags.
```

**Review by sergepavle — CHANGES_REQUESTED** [review]

File: src/Plugin/Filter/FilterOembed.php [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+      $container->get('config.factory'),
+      $container->get('renderer'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process($text, $langcode) {
```
Comment (sergepavle) [inline src/Plugin/Filter/FilterOembed.php:97]: Looks like here we don't support correctly translated version of entities (as we didn't have this in original filter for media).

File: tests/src/Functional/EmbedFilterTest.php [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+   */
+  public function testEmbedFilter(): void {
```
Comment (sergepavle) [inline tests/src/Functional/EmbedFilterTest.php:67]: Looks like we have missed test coverage for translated nodes.

File: src/Plugin/Filter/FilterOembed.php (hunk 5 above)
Comment (upchuk) [reply src/Plugin/Filter/FilterOembed.php:97]: No need. Upon rendering it renders in the current language.

File: tests/src/Functional/EmbedFilterTest.php (hunk 6 above)
Comment (upchuk) [reply tests/src/Functional/EmbedFilterTest.php:67]: It's part of the current implementation. We are not yet sure of the requirements as to whether it should render in the current site language or in the node language so we keep it as is. Once we get the requiremnt from COMM to change, we adapt

**Review by sergepavle — APPROVED** [review]

**Pull request merged**